### PR TITLE
Move schema to pkg/config

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: pkg/config/schema/generate.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Istio Authors
```
Comment (ayj) [inline pkg/config/schema/generate.go]: Any idea why this isn't showing as a file rename?

Comment (ayj) [reply pkg/config/schema/generate.go]: Maybe because the content changed as well?

**Comment by ayj** [comment]
LGTM. 

Can you add config maintainers to the codeowners file for pkg/config? Right now its only networking.

**Comment by rshriram** [comment]
Is there a strong reason to move technically? All the systems that generate Istio configs programmatically use the schema package. All of them would now have to be reworked when they update the istio dependency.

**Review by diemtvu — APPROVED** [review]

**Comment by nmittler** [comment]
@rshriram it's common code and doesn't belong in galley. This PR will contain all necessary changes in istio.

File: pkg/config/schema/generate.go (hunk 1 above)
Comment (nmittler) [reply pkg/config/schema/generate.go]: yeah not sure ... moves often confuse git in interesting ways.

**Review by ayj — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

**Comment by howardjohn** [comment]
@nmittler `FAIL	istio.io/istio/galley/pkg/config/source/kube/apiserver [build failed]`

**Comment by howardjohn** [comment]
/retest

Seems like the new test I added may be flaky, looking into it

**Pull request merged**